docs_src/design/adr/security/tbd-authentication-e2ee.md

```diff
@@ -0,0 +1,414 @@
+and the receiver has cryptographic proof of the sending service's identity.
+
+### Feature overlaps with EdgeX JWT-based authentication
+
+OpenZiti focuses on being a transport-level solution,
+controlling access to remove services.
+It does not provide application-level security.
+The use of an OpenZiti tunneler causes remote connections
+appear to come from the tunneler instead of the true client.
+In this case,
+use of JWT authentication could be used to authenticate the true client;
+for example, a `curl` client.
+
+Another case where this may apply is if an adopter elects to keep
+an NGINX API gateway as part of the solution.
+Since NGINX intercepts all requests to the backend,
+and does not itself integrate directly with OpenZiti,
+NGINX would need to use an OpenZiti tunneler to forward
+requests to backend microservices,
+and JWT authentication would flow from the remote client,
+through NGINX, through the tunneler, into the EdgeX
+microservice listening on the OpenZiti overlay.
+Without JWT authentication,
+it would appear to an EdgeX microservice that all calls
+where coming from the tunneler being used by NGINX.
+As EdgeX today does not have the notion of a "user",
+this distinction is moot, so long as NGINX validates the JWT.
+Microservice-level JWT validation can be used
+if the caller's JWT-based identity is important.
+
+
+### High-level list of changes
+
+A proof of concept implementation required the following high-level list of changes:
+
+- Add configuration options to enable use of OpenZiti. In the
+  the `Service` blocks nd `Clients.*` blocks, add a generic string map
+  named `SecurityOptions`.  OpenZiti integration will be triggered via
+  `Mode = zerotrust`.
+
+- `go-mod-bootstrap` will break out the `ListenAndServe` call into two
+  separate steps: (1) create an OpenZiti listener and (2) serve on it
+  when `Mode = zerotrust`.  Outgoing connections made in `go-mod-bootstrap`
+  will also use the OpenZiti integration.
+
+- `go-mod-bootstrap` will have the common authentication handler changed to
+  optionally accept the OpenZiti remote identity as authentication
+  in lieu of JWT authentication. JWT authentication in addition
+  to OpenZiti authentication will still be possible if needed for
+  application-level authentication.
+  
+- `go-mod-bootstrap` will open up a second unauthenticated HTTP port
+  on a separate port on the network underlay to serve the health check
+  endpoint `/api/vX/ping`.  This is required for compatibility with
+  Consul-based health checks as well as Kubernetes-based health checks.
+  This can be optionally disabled.
+  
+- `go-mod-core-contracts` will modify `makeRequest()` to use
+  OpenZiti's transport to make connections to EdgeX microservices via OpenZiti.
+
+- `edgex-ui-go`, for connections proxied via its backend,
+  will have similar adaptions as `go-mod-core-contracts`
+  when `Mode = zerotrust`.
+
+- `edgex-compose` will have added documentation that refers users
```

Comment:
Compose Builder will need new options to generate a OpenZiti enabled stack?

Reply:
Yes.